test/e2e_node/container_manager_test.go

```diff
@@ -100,7 +100,32 @@ var _ = SIGDescribe("Container Manager Misc", framework.WithSerial(), func() {
 					return validateOOMScoreAdjSetting(kubeletPids[0], -999)
 				}, 5*time.Minute, 30*time.Second).Should(gomega.BeNil())
 			})
-			ginkgo.Context("", func() {
+
+			ginkgo.Context("with test pods", func() {
+				var testPod *v1.Pod
+
+				// Log the running containers here to help debugging.
+				ginkgo.AfterEach(func(ctx context.Context) {
+					if ginkgo.CurrentSpecReport().Failed() {
+						ginkgo.By("Dump all running containers")
+						runtime, _, err := getCRIClient()
+						framework.ExpectNoError(err)
+						containers, err := runtime.ListContainers(ctx, &runtimeapi.ContainerFilter{
+							State: &runtimeapi.ContainerStateValue{
+								State: runtimeapi.ContainerState_CONTAINER_RUNNING,
+							},
+						})
+						framework.ExpectNoError(err)
+						framework.Logf("Running containers:")
```

Comment:
I am trying to remember, isn't it being dumped for failed tests already?

Reply:
not that I recall, this is why I added this code, to find out the leaked pod